### PR TITLE
fix: add License-file metadata entry

### LINKED ISSUE #328: Set `License` metadata as well
I am not sure about what magic fedora uses with [their macro](https://src.fedoraproject.org/rpms/pyproject-rpm-macros/blob/f31/f/pyproject_save_files.py), but apparently it is able to auto-detect and mark python files as license files, e.g.:
```console
$ rpm -qpL python3-packit-0.75.0-1.fc38.noarch.rpm
/usr/lib/python3.11/site-packages/packitos-0.75.0.dist-info/licenses/LICENSE
```
But this does not work here:
```console
$ rpm -qpL python3-spglib-2.1.0.rc1-1.20230518123112228603.pr272.50.ge38c0dc.x86_64.rpm 
warning: python3-spglib-2.1.0.rc1-1.20230518123112228603.pr272.50.ge38c0dc.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID bcf19e6e: NOKEY
```

My suspicion is because of the difference in `METADATA` file:
```
# packit
License-Expression: MIT
License-File: LICENSE
# spglib (where it doesn't detect)
License: BSD-3-Clause
```
Maybe the guys who showed me this feature can know more about it ([discussion](https://github.com/packit/packit/pull/1913#discussion_r1199132232))

Edit: Confirmed it is `License-File`.

### PR DESCRIPTION
**Pull request by henryiii**
Fix #328. Follows the draft of PEP 639 more closely.

### CODE REVIEW

File: tests/test_pyproject_pep517.py [hunk 1]
```diff
@@ -273,6 +273,7 @@ def test_prepare_metdata_for_build_wheel():
         "Requires-Python": ">=3.7",
         "Provides-Extra": "test",
         "Requires-Dist": 'pytest>=6.0; extra == "test"',
+        "License-File": "licenses/LICENSE",
```
Comment (LecrisUT) [inline tests/test_pyproject_pep517.py:276]: I don't think it's supposed to be prefixed with licenses

Comment (LecrisUT) [reply tests/test_pyproject_pep517.py:276]: Probably related issue: https://github.com/pypa/setuptools/issues/3596

Comment (henryiii) [reply tests/test_pyproject_pep517.py:276]: Ahh, this is https://peps.python.org/pep-0639/#license-doc-setuptools-wheel - which hasn't been accepted yet.

Comment (henryiii) [reply tests/test_pyproject_pep517.py]: Like hatchling, I'm now following the current form of the unaccepted pep. Nested files with the same name should be supported now too.

**Pull request merged**